packages/shared/src/components/modals/post/CreateSharedPostModal.tsx

```diff
@@ -53,12 +60,18 @@ export function CreateSharedPostModal({
       onRequestClose(null);
     },
   });
+
+  useEffect(() => {
+    if (isSuccess) {
+      onRequestClose(null);
+    }
+  }, [isSuccess, onRequestClose]);
```

Comment:
Maybe it's better if the `onSubmitPost` has a callback signature?
That way we can avoid the useEffect here.

Reply:
I added an onSettled here as well